hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -38,102 +41,113 @@
+				.to("log:foo").end().end()
+				.log("HNSecure received a request").setHeader("isFileDropsAllowed")
+				.simple(isFileDropsAllowed)
+				// Extract the message using custom extractor and log
+				.setBody().method(new FhirPayloadExtractor()).log("Decoded V2: ${body}")
+				//Added wireTap for asynchronous call to filedrop request
+				.wireTap("direct:start").choice().when(header("isFileDropsAllowed").isEqualToIgnoreCase(Boolean.TRUE))
+				.bean(RequestFileDropGenerater.class).id("V2FileDropsRequest").end()
+
+				// Validate the message				
+				.process(validator).id("Validator")			
+				// set the receiving app, message type into headers
+				.bean(PopulateReqHeader.class).id("PopulateReqHeader")
+				.to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
+				.log("The message receiving application is <${in.header.receivingApp}> and the message type is <${in.header.messageType}>.")
+
+				// dispatch the message based on the receiving application code and message type
+				.choice()
+				// sending message to pharmaNet
+				.when(header("receivingApp").isEqualTo(Util.RECEIVING_APP_PNP))
+				.log("Message identified as PharmaNet message. Preparing message for PharmaNet.")
+				.to("log:HttpLogger?level=DEBUG&showBody=true&multiline=true")
+				.setBody(body().regexReplaceAll("\r\n", "\r")).setBody().method(new Base64Encoder()).setBody()
+				.method(new ProcessV2ToPharmaNetJson()).id("ProcessV2ToPharmaNetJson").log("Sending to Pharmanet")
+				.removeHeader(Exchange.HTTP_URI) // clean this header as it has been set in the "from" section
+				.setHeader(Exchange.CONTENT_TYPE, constant("application/json")).setHeader(CAMEL_HTTP_METHOD, POST)
+				.setHeader(AUTHORIZATION, simple(basicToken))
+				.to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true").to(pharmNetUrl)
+				.id("ToPharmaNet").log("Received response from Pharmanet")
+				.to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
+				.process(new PharmaNetPayloadExtractor())
+
+				// sending message to HIBC for ELIG
+				.when(simple(
+						"${in.header.messageType} == {{hibc-r15-endpoint}} || ${in.header.messageType} == {{hibc-e45-endpoint}}"))
+				.log("the HIBC endpoint(${in.header.messageType}) is reached and message will be dispatched to message queue(ELIG).")
+				.setBody(simple(SampleMessages.e45ResponseMessage))
+
+				// sending message to HIBC for ENROL
+				.when(simple("${in.header.messageType} == {{hibc-r50-endpoint}}"))
+				.log("the HIBC endpoint (${in.header.messageType}) is reached and message will be dispatched to message queue(ENROL).")
+				.setBody(simple(SampleMessages.r50ResponseMessage))
+
+				// others sending to JMB
+				.otherwise().log("the JMB endpoint is reached and message will be dispatched to JMB!!")
+				.setBody(simple(SampleMessages.r03ResponseMessage)).end();
+		
+		from("direct:start").log("new route").to("log:foo").log("wire tap done");
```

Comment:
It looks like the request file drop isn't actually a part of the wiretap call. The wiretapped route here is just doing logging

Reply:
done